services/src/main/scala/cromwell/services/MetadataServiceActor.scala

```diff
@@ -34,3 +37,25 @@ object MetadataServiceActor {
   }
   */
 }
+
+object MetadataServiceActorImplicits {
+  implicit class EnhancedServiceRegistryActorForMetadata(val actor: ActorRef) extends AnyVal {
+    def pushWdlValueMetadata(metadataKey: MetadataKey, output: WdlValue): Unit = output match {
+      case WdlArray(_, valueSeq) =>
+        val zippedSeq = valueSeq.zipWithIndex
+        zippedSeq foreach { case (value, index) => actor.pushWdlValueMetadata(metadataKey.copy(key = s"${metadataKey.key}[$index]"), value) }
+      case WdlMap(_, valueMap) =>
+        valueMap foreach { case (key, value) => actor.pushWdlValueMetadata(metadataKey.copy(key = metadataKey.key + s":$key"), value) }
+      case value =>
+        actor ! PutMetadataAction(MetadataEvent(metadataKey, MetadataValue(value.valueString)))
+    }
+
+    def pushThrowableMetadata(metadataKey: MetadataKey, t: Throwable): Unit = {
+      actor ! PutMetadataAction(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:message"), MetadataValue(t.getMessage)))
+      t.getStackTrace.zipWithIndex foreach { case (trace, index) =>
+        actor ! PutMetadataAction(MetadataEvent(metadataKey.copy(key = s"${metadataKey.key}:stackTrace[$index]"), MetadataValue(trace.toString)))
```

Comment:
shouldn't the event be just a notification of the service having an error, with information we expect could be useful for some other component? and then let the stack trace go to the standard logging stash for that service?